runtime_strip.go

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) 2020, The Garble Authors.
+// See LICENSE for licensing information.
+
+package main
+
+import (
+	"go/ast"
+	"go/token"
+	"strings"
+
+	ah "mvdan.cc/garble/internal/asthelper"
+)
+
+// stripRuntime removes unnecessary code from the runtime,
+// such as panic and fatal error printing, and code that
+// prints trace/debug info of the runtime.
+func stripRuntime(filename string, file *ast.File) {
+	stripPrints := func(node ast.Node) bool {
+		call, ok := node.(*ast.CallExpr)
+		if !ok {
+			return true
+		}
+		id, ok := call.Fun.(*ast.Ident)
+		if !ok {
+			return true
+		}
+
+		switch id.Name {
+		case "print", "println":
+			id.Name = "hidePrint"
+			return false
+		default:
+			return true
+		}
+	}
+
+	switch filename {
```

Comment:
I think this would be shorter and simpler if you looped around file.Decls first, then do a type switch, then you switched on filename in there or as needed. Because practically every file has to loop over the declarations and find functions.

Reply:
Well, there are dozens of files in the runtime package, and I only modify 3 or 4 specifically, so that might be less efficient

Reply:
then you can just do `default: break` :) a loop that breaks immediately is pretty much a no-op. it's quite likely that the code duplication for the looping and type switching is hurting you more, in relative to a loop that breaks on the first iteration pretty often.